### ISSUE TITLE
providing connectAttributes and no authSwitch ( or plugin auth ) results in "Bad Handshake" error

### ISSUE THREAD

**Issue by sidorares** [sidorares/node-mysql2#611]
http://web.archive.org/web/20170404102640/https://dev.mysql.com/doc/internals/en/connection-phase-packets.html#packet-Protocol::HandshakeResponse

```
4              capability flags, CLIENT_PROTOCOL_41 always set
4              max-packet size
1              character set
string[23]     reserved (all [0])
string[NUL]    username
  if capabilities & CLIENT_PLUGIN_AUTH_LENENC_CLIENT_DATA {
lenenc-int     length of auth-response
string[n]      auth-response
  } else if capabilities & CLIENT_SECURE_CONNECTION {
1              length of auth-response
string[n]      auth-response
  } else {
string[NUL]    auth-response
  }
  if capabilities & CLIENT_CONNECT_WITH_DB {
string[NUL]    database
  }
  if capabilities & CLIENT_PLUGIN_AUTH {
string[NUL]    auth plugin name
  }
  if capabilities & CLIENT_CONNECT_ATTRS {
lenenc-int     length of all key-values
lenenc-str     key
lenenc-str     value
   if-more data in 'length of all key-values', more keys and value pairs
  }
```

looks like there is mismatch like reporting pluginAuth and not sending 'auth plugin name' ( or the other side around), need to investigate.

There is a test currently, but only for auth switch + connection attributes, not for initially connecting with attributes - https://github.com/sidorares/node-mysql2/blob/5f0fb8f1f5035e2c0207490aa2f0b838dc82fdc2/test/integration/test-auth-switch.js

sql to see connect attributes for session: `SELECT * FROM performance_schema.session_connect_attrs WHERE processlist_id = CONNECTION_ID()`

also some info here https://github.com/mysqljs/mysql/issues/1761

**Comment by sidorares** [comment]
as part of this issue - need to document api, looks like nobody knows about it

**Comment by sidorares** [comment]
can't reproduce anymore, likely fixed as a side effect of #1062
